packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -567,6 +577,7 @@ class _DropdownRoute<T> extends PopupRoute<_DropdownRouteResult<T>> {
       scrollOffset = math.min(scrollOffset, preferredMenuHeight - menuHeight);
     }
 
+    assert(menuHeight == menuBottom - menuTop);
```

Comment:
Can you add a test that triggers this assertion?

Reply:
If we only add this assertion to the master branch, there will be many tests that trigger this assertion.
 Fortunately, no one has used `_MenuLimits.menuBottom` before, so there are no functional issues.

Reply:
This is just to check that the metrics calculated by this function are correct. 
I didn't think of a good way to test and trigger it.